src/main/java/com/aws/greengrass/clientdevices/auth/metrics/MetricsEmitter.java

```diff
@@ -30,15 +29,31 @@ public MetricsEmitter(ScheduledExecutorService ses, ClientDeviceAuthMetrics metr
     }
 
     /**
-     * Start emitting metrics with no initial delay.
+     * Restart the metrics emitter and apply configurations.
+     *
+     * @param disableMetrics               Enable metrics flag
+     * @param periodicAggregateIntervalSec Periodic aggregate interval in seconds
      */
-    public void start() {
+    public void restart(boolean disableMetrics, int periodicAggregateIntervalSec) {
+        if (disableMetrics) {
+            return;
```

Comment:
i think we're never stopping metrics emitter if we disable it in config?

Reply:
Oh that's true, I need to stop it before returning